### ISSUE TITLE
Behavior change after recent commit (passive progress)

### ISSUE THREAD

**Issue by tonycurtis** [openucx/ucx#4276]
Until recently, I was able to get passive progress, e.g. as in

https://github.com/openshmem-org/openshmem-examples/blob/master/C/ptp.c

where one rank/PE makes no SHMEM calls.

Since commit 7dc71b3a9b70a0ca3a6e2fb36b67466d0eac934b, the program now hangs.

Interconnect is

```shell
$ lspci | grep -i band
06:00.0 Infiniband controller: Mellanox Technologies MT27600 [Connect-IB]
```

(mlx5 driver).  Same thing seen on another IB cluster.  With an added progress thread it works (as for older MLNX IB mlx4)

**Comment by brminich** [comment]
what is your command line? Can you please post output of UCX_LOG_LEVEL=info?

**Comment by tonycurtis** [comment]
Should see

```shell
$ oshcc ptp.c
$ oshrun -n 4 ./a.out
[1] elapsed usecs 31 A 0
[2] elapsed usecs 4561 A 0
[3] elapsed usecs 4564 A 0
[0] elapsed usecs 4652 A 2
$
```

or similar.  buggy behavior is

```shell
$ oshcc ptp.c
$ oshrun -n 4 ./a.out
[1] elapsed usecs 31 A 0
<hang>
```

Here's the LOG_LEVEL output

```shell
[1570811998.112739] [login2:49973:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[1]: rma(self/memory posix/memory sysv/memory);
[1570811998.112759] [login2:49975:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[1]: rma(rc_mlx5/mlx5_0:1 posix/memory sysv/memory);
[1570811998.112779] [login2:49976:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[1]: rma(rc_mlx5/mlx5_0:1 posix/memory sysv/memory);
[1570811998.112783] [login2:49974:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[1]: rma(rc_mlx5/mlx5_0:1 posix/memory sysv/memory);
[1570811998.112976] [login2:49973:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[2]: rma(rc_mlx5/mlx5_0:1 posix/memory sysv/memory);
[1570811998.130362] [login2:49974:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[2]: rma(self/memory posix/memory sysv/memory);
[1570811998.138320] [login2:49976:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[2]: rma(self/memory posix/memory sysv/memory);
[1570811998.142419] [login2:49975:0]     ucp_worker.c:1516 UCX  INFO  ep_cfg[2]: rma(self/memory posix/memory sysv/memory);
[1] elapsed usecs 5 A 0
```

Same log output in both cases

**Comment by brminich** [comment]
thanks, checking

**Comment by tonycurtis** [comment]
just updated the log messages

**Comment by brminich** [comment]
it was broken by c0e8ee1
fixed in #4277

**Comment by tonycurtis** [comment]
Just patched with the PR, it's working again.  thanks.

**Comment by evgeny-leksikov** [comment]
> it was broken by [c0e8ee1](https://github.com/openucx/ucx/commit/c0e8ee126d970e88127d06bd01477f88b631574a)
> fixed in #4277

@brminich maybe make sense to change `UCP_ADDRESS_PACK_FLAG_DISABLE_HW_AMO` to `UCP_ADDRESS_PACK_FLAG_ENABLE_HW_AMO`? it would make possible to pass `-1` by default as before
